lib/bundler/resolver.rb

```diff
@@ -267,7 +268,9 @@ def search_for(dependency)
           []
         end
       end
-      search.select {|sg| sg.for?(platform, @ruby_version) }.each {|sg| sg.activate_platform!(platform) }
+      platform_results = search.select {|sg| sg.for?(platform, @ruby_version) }.each {|sg| sg.activate_platform!(platform) }
+      return platform_results if @gem_version_promoter.level == :major # default behavior
+      @gem_version_promoter.sort_versions(dependency, platform_results)
```

Comment:
most likely. part of this is leftovers from how i had to hack on this from the outside in bundler-patch - part of this was due to a weird bug I never chased down where on large resolves sometimes bundler clears the result array in-place, corrupting the cache. it never matters in bundler due to the pre-existing search.select effectively duping the result ... but it mattered in bundler-patch which worksaround it by duping its result (here in the last line of `#sort_versions`). 

I did have an idea that freezing the result could flush out the (presumably) offending code, and then, yeah, the whole thing could go inside. Or the whole thing could go inside and then we #dup once right here.